content/en/user-guide/aws/cognito/index.md

```diff
@@ -331,6 +331,69 @@ Ensuring this match is crucial for the proper functioning of the authentication
 {"access_token": "eyJ0eXAi…lKaHx44Q", "expires_in": 86400, "token_type": "Bearer", "refresh_token": "e3f08304", "id_token": "eyJ0eXAi…ADTXv5mA"}
 ```
 
+### Client credentials grant
+
+The client credentials grant is designed for machine-to-machine (M2M) communication.
+In contrast, the authorization code and implicit grants provide tokens to authenticated human users.
+The client credentials grant allows for scope-based authorization from a non-interactive system to an API.
+Your app can directly request client credentials from the token endpoint to receive an access token.
+
+To request the token from LocalStack the correct URL is `http://cognito-idp.localhost.localstack-test.cloud:4566/_aws/cognito-idp/oauth2/token`.
```

Comment:
```suggestion
To request the token from LocalStack the correct URL is `http://cognito-idp.localhost.localstack.cloud:4566/_aws/cognito-idp/oauth2/token`.
```

I think that you copy-pasted this from my issue, but the `-test` part was relevant for me because I had 2 LocalStack containers:
- `localstack`: for manual testing
- `localstack-test`: for automatic tests

Also, I have to say that the URL that worked for me was the following one:
```python
token_url = "http://localhost.localstack-test.cloud:4566/_aws/cognito-idp/oauth2/token"
```

However, I also had the domain configured for the user pool:
```sh
awslocal cognito-idp create-user-pool-domain \
  --domain "${COGNITO_USER_POOL_DOMAIN_PREFIX}" \
  --user-pool-id "${COGNITO_USER_POOL_ID}" \
  --region "${COGNITO_AUTH_REGION}"
```

Yet, supposedly, when a Domain is configured, the token should be accesible from another endpoint:
```
"https://mydomain.auth.us-east-1.amazoncognito.com/_aws/cognito-idp/oauth2/token"
```
(Doubt: in AWS, if the domain is configured are endpoints available or just the domain one?)

Based on: https://docs.aws.amazon.com/cognito/latest/developerguide/token-endpoint.html#post-token-positive 
> The token endpoint becomes publicly available when you add a domain to your user pool. It accepts HTTP POST requests. For application security, use PKCE with your authorization code sign-in events. PKCE verifies that the user passing an authorization code is that same user who authenticated. For more information about PKCE, see [IETF RFC 7636](https://datatracker.ietf.org/doc/html/rfc7636).

But I don't know if LocalStack is capable of replicating this behavior (due to being local and requiring HTTPS).